### DISCUSSION TITLE
Notification in MQTT On Fatal Errors

### DISCUSSION THREAD

**Discussion by gniknalu** [roflcoopter/viseron#254] [Q&A] [answered]
How can I get a MQTT notification when there are fatal FFMpeg errors like "Connection refused" or "No route to host"? That would allow me to monitor for repeated errors and know if a camera is off-line (or if they're all offline).

Thank you

**Comment by roflcoopter** [comment, marked as answer]
Not possible right now sadly, but it is on my to do list!

Right now development has stalled a little since i just moved to a house and there is a lot of things that i have to setup.
I hope to pick up development in a few weeks, and this one should be easy to implement.

Sorry for the delay!

**Reply by gniknalu** [reply]
> Right now development has stalled a little since i just moved to a house and there is a lot of things that i have to setup.
> I hope to pick up development in a few weeks, and this one should be easy to implement.

No Worries. 

**Congratulations on your new place!**

Thank you!

**Comment by roflcoopter** [comment]
This is now implemented in the `dev` branch.
The MQTT status sensor will now show `disconnected` when the camera cannot be reached.

Sorry it took me so long!

**Reply by gniknalu** [reply]
> This is now implemented in the `dev` branch.
> The MQTT status sensor will now show `disconnected` when the camera cannot be reached.

Your AWESOME. **Thank you!** Will load it right now!

**Reply by gniknalu** [reply]
> > This is now implemented in the `dev` branch.
> > The MQTT status sensor will now show `disconnected` when the camera cannot be reached.
> 
> Your AWESOME. **Thank you!** Will load it right now!

With the dev branch loaded, I'm getting this error

```
[2021-09-23 09:08:29] [INFO    ] [viseron                 ] - -------------------------------------------
[2021-09-23 09:08:29] [INFO    ] [viseron                 ] - Initializing...
[2021-09-23 09:08:30] [INFO    ] [viseron.mqtt            ] - Initializing MQTT connection
[2021-09-23 09:08:30] [INFO    ] [viseron                 ] - Initializing NVR threads
OpenCV(ocl4dnn): consider to specify kernel configuration cache directory
                 via OPENCV_OCL4DNN_CONFIG_PATH parameter.
OpenCL program build log: dnn/dummy
Status -11: CL_BUILD_PROGRAM_FAILURE
-cl-no-subgroup-ifp
Error in processing command line: Don't understand command line argument "-cl-no-subgroup-ifp"!
```

**Reply by roflcoopter** [reply]
Hmm that seems like a driver issue. Is your machine updated?

**Reply by gniknalu** [reply]
> Hmm that seems like a driver issue. Is your machine updated?

Everything was running fine - using the dev version with the updated CUDA 11.4.1 since you updated it in the dev branch.

**Reply by roflcoopter** [reply]
Do you have multiple Viseron containers running simultaneously?

**Reply by gniknalu** [reply]
> Do you have multiple Viseron containers running simultaneously?

No. In fact I deleted the previous container first. 
On the other machine - the one with the problems - it has the same issue. Prior to loading the dev branch there were no containers in docker.

**Reply by roflcoopter** [reply]
Damn it 😂
Something always goes wrong doesnt it?

**Reply by gniknalu** [reply]
> Damn it 😂
> Something always goes wrong doesnt it?

That's what I get for testing in production!  😂😂

**Reply by gniknalu** [reply]
> Damn it 😂
> Something always goes wrong doesnt it?

If I'm the only one with this error, I'll go through all my nvidia drivers to see what is wrong. (but will wait first to see if anyone else reports the error.)

**Reply by roflcoopter** [reply]
There is some information here: https://github.com/opencv/opencv/issues/14187

Seems you can ignore the error, its just a warning that a specific OpenCL function is not supported on the hardware.
I have no explanation tho why it worked before...

**Reply by gniknalu** [reply]
> There is some information here: [opencv/opencv#14187](https://github.com/opencv/opencv/issues/14187)
> 
> Seems you can ignore the error, its just a warning that a specific OpenCL function is not supported on the hardware.
> I have no explanation tho why it worked before...

It's bad enough to stop viseron from working - on both computers the same behavior is observed. I reverted to the 'latest' build since the dev build didn't stopped functioning. I turned on debug logging to see if I could get extended information and didn't get any more details.

**Reply by gniknalu** [reply]
Now that the OpenCL error isn't fatal in the `dev` pull, I can test this.

**Comment by gniknalu** [comment]
Works Great! **Thank you!**

This is going to help a lot!

**Comment by roflcoopter** [comment]
Wohoo, happy to hear!
